example/lib/peripheral_details/components/peripheral_details_view.dart

```diff
@@ -29,4 +38,99 @@ class PeripheralDetailsView extends StatelessWidget {
       ],
     );
   }
+
+  Widget _createServiceView(
+    BuildContext context,
+    PeripheralDetailsState state,
+  ) {
+    return Flexible(
+      fit: FlexFit.loose,
+      child: ListView.builder(
+        shrinkWrap: true,
+        itemCount: state.bleServiceStates.length,
+        itemBuilder: (context, index) =>
+            _createServiceTileView(context, state.bleServiceStates[index]),
+      ),
+    );
+  }
+
+  Widget _createServiceTileView(
+    BuildContext context,
+    BleServiceState serviceState,
+  ) {
+    // ignore: close_sinks
+    final PeripheralDetailsBloc bloc =
+        BlocProvider.of<PeripheralDetailsBloc>(context);
+
+    return Column(
+      children: <Widget>[
+        PropertyRow(
+          title: "Service UUID",
+          titleColor: Theme.of(context).primaryColor,
+          value: serviceState.service.uuid,
+          valueTextStyle: CustomTextStyle.serviceUuidStyle,
+          rowAccessory: IconButton(
+            icon: Icon(
+                serviceState.expanded ? Icons.unfold_less : Icons.unfold_more),
+            onPressed: () => bloc.add(ServiceViewExpandedEvent(
+              serviceState,
+              !serviceState.expanded,
+            )),
+          ),
+        ),
+        serviceState.expanded
```

Comment:
correct me if I am wrong, at the moment if a single service is supposed to be expanded the whole view is redrawn. it seems to be not ok.

Reply:
As far as I know Flutter handles it in a way that redraws only the changed part of the view. I consulted it with @mikolak and it seems to be ok.